### PR TITLE
Server Testsuite changes for Native Image

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-11850
https://issues.redhat.com/browse/ISPN-11936
https://issues.redhat.com/browse/ISPN-11948

ISPN-11850 should remain open as there are still tests that won't work with the CONTAINER driver. These will be fixed at a later date.

### CODE REVIEW

File: server/tests/src/test/java/org/infinispan/server/cli/CliIT.java [hunk 1]
```diff
@@ -94,25 +101,46 @@ public void testCliInteractive() {
...
-      shell.assertContains("batch1");
+   public void testCliBatch() throws IOException {
+      // Create batch file programmatically so that we can supply the correct server address for the connect command:
+      Path testPath = Paths.get(CommonsTestingUtil.tmpDirectory(CliIT.class));
+      testPath.toFile().mkdirs();
+      try {
+         Path batchFilePath = testPath.resolve("batch.cli").toAbsolutePath();
+         try (PrintWriter out = new PrintWriter(Files.newBufferedWriter(batchFilePath))) {
+            out.println("echo Hi CLI");
+            out.printf("connect %s:11222\n", hostAddress());
```
Comment (tristantarrant) [inline server/tests/src/test/java/org/infinispan/server/cli/CliIT.java:112]: Interesting. Makes me think we should support env and property replacement in batches

**Comment by ryanemerson** [comment]
```
org.infinispan.server.extensions.ServerTasks.testDistributedServerTaskWithParameters
org.infinispan.server.functional.RestLoggingResource.testManipulateLogger
``
Failures are related :eyes:

File: server/tests/src/test/java/org/infinispan/server/cli/CliIT.java (hunk 1 above)
Comment (ryanemerson) [reply server/tests/src/test/java/org/infinispan/server/cli/CliIT.java:112]: That would be great :slightly_smiling_face:

File: server/testdriver/core/src/main/java/org/infinispan/server/test/api/HotRodTestClientDriver.java [hunk 2]
```diff
@@ -31,18 +33,32 @@ public HotRodTestClientDriver(TestServer testServer, TestClient testClient) {
...
+      return this;
+   }
+
+   /**
+    * The {@link Marshaller} to be used by the client.
+    *
+    * @param marshallerClass
+    * @return the current {@link HotRodTestClientDriver} instance with the Marshaller configuration override
+    */
+   public HotRodTestClientDriver withMarshaller(Class<? extends Marshaller> marshallerClass) {
```
Comment (ryanemerson) [inline server/testdriver/core/src/main/java/org/infinispan/server/test/api/HotRodTestClientDriver.java:60]: @tristantarrant I couldn't think of a better way of accomplishing this whilst keeping `infinispan-jboss-marshalling` on the classpath and making it so by default the tests utilise ProtoStream. In 12 we should probably remove the automatic configuration of `GenericJbossMarshaller` when `infinispan-jboss-marshalling` is present on the classpath.

File: server/tests/src/test/java/org/infinispan/server/cli/CliIT.java (hunk 1 above)
Comment (tristantarrant) [reply server/tests/src/test/java/org/infinispan/server/cli/CliIT.java:112]: https://github.com/infinispan/infinispan/pull/8435

Comment (ryanemerson) [reply server/tests/src/test/java/org/infinispan/server/cli/CliIT.java]: Rebased on your commit. Much simpler

File: server/testdriver/core/src/main/java/org/infinispan/server/test/api/HotRodTestClientDriver.java (hunk 2 above)
Comment (ryanemerson) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/api/HotRodTestClientDriver.java:60]: https://issues.redhat.com/browse/ISPN-11974

**Comment by tristantarrant** [comment]
https://ci.infinispan.org/job/Infinispan/job/PR-8430/5/testReport/junit/org.infinispan.server.extensions/ScriptingTasks/testStreamingScript/ seems related

**Comment by ryanemerson** [comment]
Yep. I've fixed locally, just updating `RollingUpgradeIT` to work with containers and I'll repush

**Comment by ryanemerson** [comment]
Failures unrelated

**Comment by tristantarrant** [comment]
Merged

**Pull request merged**